src/client/address_book/storage.cc

```diff
@@ -68,15 +69,20 @@ bool AddressBookStorage::GetAddress(
 void AddressBookStorage::AddAddress(
     const kovri::core::IdentityEx& address) {
   auto filename = GetAddressesPath() / (address.GetIdentHash().ToBase32() + ".b32");
-  std::ofstream file(filename.string(), std::ofstream::binary);
-  if (!file)
+  kovri::core::OutputFileStream file(filename.string(), std::ofstream::binary);
+  if (file.Bad() || file.Fail())
     LOG(error) << "AddressBookStorage: can't open file " << filename;
```

Comment:
If we can't open a file for writing, then there's no point in trying to write below. This should either throw or return (ideally throw).